sentry-core/src/main/java/io/sentry/core/transport/HttpTransport.java

```diff
@@ -292,14 +301,20 @@ private void updateRetryAfterLimits(
           if (retryAfterAndCategories.length > 1) {
             final String allCategories = retryAfterAndCategories[1];
 
-            if (allCategories != null) {
+            if (allCategories != null && !allCategories.isEmpty()) {
               final String[] categories = allCategories.split(";", -1);
 
               for (final String catItem : categories) {
                 // we dont care if Date is UTC as we just add the relative seconds
                 sentryRetryAfterLimit.put(
                     catItem, new Date(System.currentTimeMillis() + retryAfterMillis));
               }
+            } else {
+              // if categories are empty, we should apply to all the categories.
+              // we do that using a `default` category as the fallback
+              sentryRetryAfterLimit.clear();
```

Comment:
another way would be to loop all the items of sentryRetryAfterLimit and replace them.

Reply:
This sounds better to me too